### ISSUE TITLE
Rule idea: `declaration-prefer-shorthand`

### ISSUE THREAD

**Issue by silverwind** [stylelint/stylelint#7686]
### What is the problem you're trying to solve?

It should be completely safe to reduce cases of longhands with equal values to their corresponding shorthands:

```css
.foo {
  left: 0; 
  right: 0;
  top: 0;
  bottom: 0;
  margin-top: 0;
  margin-bottom: 0;
  margin-left: 4px;
  margin-right: 4px;
}
```

### Pass

```css
.foo {
  inset: 0;
  margin: 0 4px;
}
```

### What solution would you like to see?

A new rule that flags unnecessary longhands of `inset`, `margin` and `padding` and more.

An argument could be made that this is a stylistic rule, but I think it's in the same category as [shorthand-property-no-redundant-values](https://stylelint.io/user-guide/rules/shorthand-property-no-redundant-values), which essentially is also stylistic.

**Comment by Mouvedia** [comment]
Are you talking about the existing [`declaration-block-no-redundant-longhand-properties`](https://stylelint.io/user-guide/rules/declaration-block-no-redundant-longhand-properties)?

**Comment by silverwind** [comment]
Indeed that seems to be exactly what I need, somehow I missed it :)
